### PR TITLE
Feature/sharder get scstate test and docs review

### PR DESCRIPTION
**Pull request by YarikRevich**
Refers to issue #373

### CODE REVIEW

File: tests/api_tests/get_scstate_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+		require.Equal(t, util.HttpOkStatus, restyResponse.Status())
+		require.Equal(t, util.ZcnToInt(sharderSCStateResponse.Used), int64(1), "SCState does not seem to be valid")
+	})
+
+	t.Run("Get SCState of faucet SC, shouldn't work", func(t *testing.T) {
+		t.Parallel()
+
+		registeredWallet, _ := registerWallet(t)
+
+		_, restyResponse, err := v1SharderGetSCState(t, FAUCET_SMART_CONTRACT_ADDRESS, registeredWallet.Id, nil)
```
Comment (stewartie4) [inline tests/api_tests/get_scstate_test.go:29]: why does this endpoint require for data as opposed to a regular JSON body?

File: internal/api/util/zerochain.go [hunk 2]
```diff
@@ -94,6 +94,34 @@ func (z *Zerochain) PostToMiner(t *testing.T, miner, endpoint string, body inter
...
 }
 
+func (z *Zerochain) PostToShardersWithFormData(t *testing.T, endpoint string, consensusMet ConsensusMetFunction, formData map[string]string, body interface{}, targetObject interface{}) (*resty.Response, error) { //nolint
+	postToSharder := func(sharder string) (*resty.Response, error) {
+		return z.PostToSharder(t, sharder, endpoint, formData, body, targetObject)
+	}
+	return z.executeWithConsensus(t, z.Sharders, postToSharder, targetObject, consensusMet)
+}
+
+func (z *Zerochain) PostToSharder(t *testing.T, sharder, endpoint string, formData map[string]string, body interface{}, targetObject interface{}) (*resty.Response, error) { //nolint
```
Comment (stewartie4) [inline internal/api/util/zerochain.go:104]: PostToSharderWithFormData

File: tests/api_tests/get_scstate_test.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+		registeredWallet, keyPair := registerWallet(t)
+		executeFaucet(t, registeredWallet, keyPair)
+
+		sharderSCStateResponse, restyResponse, err := v1SharderGetSCState(t, FAUCET_SMART_CONTRACT_ADDRESS, registeredWallet.Id, nil)
+		require.Nil(t, err)
+		require.Equal(t, util.HttpOkStatus, restyResponse.Status())
+		require.Equal(t, util.ZcnToInt(sharderSCStateResponse.Used), int64(1), "SCState does not seem to be valid")
+	})
+
+	t.Run("Get SCState of faucet SC, shouldn't work", func(t *testing.T) {
```
Comment (stewartie4) [inline tests/api_tests/get_scstate_test.go:24]: shouldn't work in what case?

File: tests/api_tests/get_scstate_test.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"github.com/0chain/system_test/internal/api/util"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestGetScState(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Get SCState of faucet SC, should work", func(t *testing.T) {
```
Comment (stewartie4) [inline tests/api_tests/get_scstate_test.go:12]: Are there other smart contracts we could also test this on? 
Faucet isn't very important as opposed to say, storage

File: tests/api_tests/get_scstate_test.go (hunk 1 above)
Comment (YarikRevich) [reply tests/api_tests/get_scstate_test.go:29]: Fixed

File: internal/api/util/zerochain.go (hunk 2 above)
Comment (YarikRevich) [reply internal/api/util/zerochain.go:104]: To tell you the truth, in general, the architecture of test suit is really bad, I'll fix that in my refactiring issue. But for now, let it be just "PostToSharder"

File: tests/api_tests/get_scstate_test.go (hunk 3 above)
Comment (YarikRevich) [reply tests/api_tests/get_scstate_test.go:24]: Fixed

File: tests/api_tests/get_scstate_test.go (hunk 4 above)
Comment (YarikRevich) [reply tests/api_tests/get_scstate_test.go:12]: It works only for faucet

**Pull request closed without merging**